modules/core/src/core/lib/pick-layers.js

```diff
@@ -61,43 +62,69 @@ export function pickObject(
     deviceHeight: pickingFBO.height
   });
 
-  const pickedColors =
-    deviceRect &&
-    drawAndSamplePickingBuffer(gl, {
+  // reset buffers
+  layers.forEach(l => l.updateInstancePickingColors([]));
```

Comment:
Shouldn't this only be done on the matching layer? Aren't we clearing other valid objects that happen to have the same index?

Nit: Type out: `l` => `layer`

Reply:
Isn't this called on every pointer move event? What is the perf implication?